### DISCUSSION TITLE
Jkbms Parameters CCL i DCL - limit cannot be changed

### DISCUSSION THREAD

**Discussion by m4in77** [Louisvdw/dbus-serialbattery#331] [Q&A] [answered]
![JK_para](https://user-images.githubusercontent.com/118193235/206706388-4b2cb370-6da3-481e-b0ef-7f0562131ac8.JPG)
Hi. I change CCL and DCL in utils.py, unfortunately in parameters - remote control remain unchanged. Other parameters like cell voltage change as can be seen from the CVL. In utils.py I changed to other values but still got 60A.
![JK_utils](https://user-images.githubusercontent.com/118193235/206706473-9dccf5ce-824f-4b83-9c38-576119e854af.JPG)
Maybe it needs to be changed somewhere else ?

**Comment by Louisvdw** [comment]
Sometimes you just need another reboot to apply the values, but also check that you do not have CCCM_T_ENABLE or CCCM_CV_ENABLE true as any oth those will limit the current using their parameters as well

**Comment by m4in77** [comment]
I checked these parameters and changed as you suggested. Now CCL is as I set in utils.py , but DCL still only 60A
![JK_DCL](https://user-images.githubusercontent.com/118193235/206710510-bedb29f4-6ff5-4940-af41-76187fde8c2e.JPG)

**Comment by Louisvdw** [comment]
Sorry, I was not very clear. There are settings to enable the charging contols seperately for Charge and Discharge. You will need to disable for discharging as well.

```
CCCM_T_ENABLE 
CCCM_CV_ENABLE
CCCM_SOC_ENABLE
```

```
DCCM_T_ENABLE 
DCCM_CV_ENABLE
DCCM_SOC_ENABLE
```

**Reply by Markus1301** [reply]
Hi Louis,
I ran with my Daly BMS in the same situation: SerialBattery(Daly) => Parameter:
CVL: 55,20V
CCL: 60A
DCL: 60A

I had changed in utils.py
     
MAX_BATTERY_CHARGE_CURRENT = 70.0                              
MAX_BATTERY_DISCHARGE_CURRENT = 180.0  

All of the 6 "flags" above were true. Now I got it: I thought, I want (!) the management. Now with some trying I got it: When I change the discharge current to 180A I have to set all 3 of DCCM_T/CV/SOC_ENABLE to  False.

You should add this in the setup instructions. Only MAX_BATTERY_* is not enough.

Markus

**Comment by m4in77** [comment]
actually, I already figured it out :) Thank you, it's working now.

**Comment by erichabg** [comment]
I runned into the same problem.

what do you think of this solution?

move the lines:
MAX_BATTERY_CHARGE_CURRENT = 100.0
MAX_BATTERY_DISCHARGE_CURRENT = 100.0

to the top of the file
and then declare the field accourding to the variables
MAX_BATTERY_CHARGE_CURRENT = 100.0
MAX_BATTERY_DISCHARGE_CURRENT = 100.0

CELL_VOLTAGES_WHILE_CHARGING         = [3.55, 3.50, 3.45, 3.30]
MAX_CHARGE_CURRENT_CV                = [   0,    2,  MAX_BATTERY_CHARGE_CURRENT/2,  MAX_BATTERY_CHARGE_CURRENT]

CELL_VOLTAGES_WHILE_DISCHARGING      = [2.70, 2.80, 2.90, 3.10]
MAX_DISCHARGE_CURRENT_CV             = [   0,    5,  MAX_BATTERY_DISCHARGE_CURRENT,  MAX_BATTERY_DISCHARGE_CURRENT]

TEMPERATURE_LIMITS_WHILE_CHARGING    = [55, 40,  35,   5,  2, 0]
MAX_CHARGE_CURRENT_T                 = [ 0, MAX_BATTERY_CHARGE_CURRENT/2, MAX_BATTERY_CHARGE_CURRENT, MAX_BATTERY_CHARGE_CURRENT, MAX_BATTERY_CHARGE_CURRENT/2, 0]

TEMPERATURE_LIMITS_WHILE_DISCHARGING = [55, 40,  35,   5,  0, -20]
MAX_DISCHARGE_CURRENT_T              = [ 0, MAX_BATTERY_DISCHARGE_CURRENT/2, MAX_BATTERY_DISCHARGE_CURRENT, MAX_BATTERY_DISCHARGE_CURRENT, MAX_BATTERY_DISCHARGE_CURRENT/2,   0]

**Reply by entropia1ac** [reply]
That's a good idea - will test it.

**Reply by vincegod** [reply]
please see https://github.com/Louisvdw/dbus-serialbattery/pull/389#issue-1519709515

**Comment by m4in77** [comment]
Hi, after installing the new version. I can't change the CVL. I have 55.2v now, how to change it?
![jkcvl](https://user-images.githubusercontent.com/118193235/220560046-75d1c115-c73e-44ac-ad76-6fe3eed59805.JPG)

**Reply by mr-manuel** [reply]
Hi, there was no new version released since v0.14.3. What is the new version for you? :)

**Reply by m4in77** [reply]
OK ;) I had to reinstall everything from scratch. As I did before, I had no problem changing it. And it seems to me that the settings looked different. Unless I don't remember :) But I can't change it anyway.

**Reply by m4in77** [reply]
I installed the driver again and now everything can be changed.

**Comment by mr-manuel** [comment, marked as answer]
See FAQ https://github.com/Louisvdw/dbus-serialbattery/wiki/FAQ#why-is-the-chargingdischarging-current-limit-ccldcl-smaller-than-the-set-one
